org.scala-ide.sdt.core/src/org/scalaide/core/internal/statistics/Statistics.scala

```diff
@@ -0,0 +1,290 @@
+package org.scalaide.core.internal.statistics
+
+import java.io.File
+import java.io.FileWriter
+
+import scala.pickling.FastTypeTag
+import scala.pickling.PBuilder
+import scala.pickling.PReader
+import scala.pickling.Pickler
+import scala.pickling.Unpickler
+import scala.pickling.pickler.AllPicklers
+
+import org.scalaide.util.eclipse.EclipseUtils
+import org.scalaide.core.internal.statistics.Features.CopyQualifiedName
+import org.scalaide.core.internal.statistics.Features.ExplicitReturnType
+import org.scalaide.core.ScalaIdeDataStore
+import Features._
+
+class Statistics {
+  import scala.pickling.Defaults._
+  import scala.pickling.json._
+
+  private var firstStat = 0L
+  private var cache = Map[Feature, FeatureData]()
+
+  readStats()
+
+  def data: Seq[FeatureData] = cache.values.toList
+  def startOfStats: Long = firstStat
+
+  def incUses(feature: Feature, numToInc: Int = 1): Unit = {
+    val stat = cache.get(feature).getOrElse(FeatureData(feature, 0, System.currentTimeMillis))
+    cache += feature → stat.copy(nrOfUses = stat.nrOfUses + numToInc, lastUsed = System.currentTimeMillis)
```

Comment:
Could you add some domain method to `FeatureData`? like `def increase(by: Long, lastUsed: Long = System.currentTimeMillis)`?